core/src/display_object/edit_text.rs

```diff
@@ -1476,36 +1491,39 @@ impl<'gc> EditText<'gc> {
             let mut changed = false;
             let mut cancelled = false;
             if !character.is_control() && self.available_chars() > 0 {
-                if let Avm2Value::Object(target) = self.object2() {
-                    let character_string =
-                        AvmString::new_utf8(context.gc_context, character.to_string());
-
-                    let mut activation = Avm2Activation::from_nothing(context.reborrow());
-                    let text_evt = Avm2EventObject::text_event(
-                        &mut activation,
-                        "textInput",
-                        character_string,
-                        true,
-                        true,
-                    );
-                    Avm2::dispatch_event(&mut activation.context, text_evt, target);
+                let allowed = self.0.read().restrict.to_allowed(character);
+                if let Some(character) = allowed {
+                    if let Avm2Value::Object(target) = self.object2() {
```

Comment:
Can't these two `if let`s be merged somehow? Not sure, nor if that would be more readable, it's just that things are getting more and more indenty here.

Reply:
Things were very indenty here, I managed to reduce the indent from 4 to mostly 1.